genai-perf/genai_perf/inputs/converters/openai_completions_converter.py

```diff
@@ -25,95 +25,52 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 import random
-from copy import deepcopy
-from typing import Dict, List
+from typing import Any, Dict
 
 from genai_perf.inputs.converters.base_converter import BaseConverter
-from genai_perf.inputs.input_constants import (
-    DEFAULT_OUTPUT_TOKENS_MEAN,
-    EMPTY_JSON_IN_OPENAI_PA_FORMAT,
-)
+from genai_perf.inputs.input_constants import DEFAULT_OUTPUT_TOKENS_MEAN
 from genai_perf.inputs.inputs_config import InputsConfig
 
 
 class OpenAICompletionsConverter(BaseConverter):
-    def convert(
-        self,
-        generic_dataset: Dict,
-        config: InputsConfig,
-    ) -> Dict:
-        (
-            system_role_headers,
-            user_role_headers,
-            text_input_headers,
-        ) = self._determine_json_feature_roles(generic_dataset)
 
-        pa_json = self._populate_openai_completions_output_json(
-            generic_dataset,
-            system_role_headers,
-            user_role_headers,
-            text_input_headers,
-            config,
-        )
+    # TODO (TPA-430): This works for great for synthetic and input file approaches
+    # but a bit tedious for dataset case as we need to specify the content names
+    # for each dataset. This is because a dataset can be used differently depending
+    # on the endpoint (e.g. chat vs non-chat).
```

Comment:
In order to remove chat API specific logics out from the non-chat APIs, these three non-chat converters need to specify the list of content keys/names it supports. This is mainly due to the complexity of supporting arbitrary public datasets. For example, each dataset comes with unique content names (e.g. "system_prompt", "question", and "response" in openorca). And depending on the endpoint, how we want to construct the payload changes. For chat endpoints, it makes sense to distinguish each contents from openorca dataset because it understands system prompt concept. However, in non-chat LLM API, there's no difference between the "system_prompt" content and "question" content in the openorca dataset, so we just simply concatenate them, but at the same time, we need to _ignore_ "response" content, hence specifying only "system_prompt" and "question" content names under `_CONTENT_NAMES` attribute. 

We probably want more scalable approach if we want to support arbitrary datasets, but that is likely out of scope of this PR. Also, given that the dataset approach is the least used feature in genai-perf, this is probably not an urgent issue.